File: find-all-numbers-disappeared-in-array/find-all-numbers-disappeared-in-array.js

var findDisappearedNumbers = function (nums) {
    for (let i = 0; i < nums.length; i++) {
        nums[Math.abs(nums[i]) - 1] = -Math.abs(nums[Math.abs(nums[i]) - 1])
    }

    ans = []
    for (let i = 0; i < nums.length; i++) {
        if (nums[i] > 0) {
            ans.push(i + 1)
        }
    }
    return ans
};

File: find-all-numbers-disappeared-in-array/find-all-numbers-disappeared-in-array.cpp
class Solution
{
public:
    vector<int> findDisappearedNumbers(vector<int> &a)
    {
        int n = a.size();

        for (int i = 0; i < n; i++)
        {
            a[abs(a[i]) - 1] = -abs(a[abs(a[i]) - 1]);
        }

        vector<int> ans;
        for (int i = 0; i < n; i++)
        {
            if (a[i] > 0)
                ans.push_back(i + 1);
        }
        return ans;
    };